modules/generator/processor/hostinfo/processor.go

```diff
@@ -0,0 +1,71 @@
+package hostinfo
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/tempo/modules/generator/registry"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"go.opentelemetry.io/otel"
+)
+
+const (
+	Name = "host-info"
+
+	hostInfoMetric     = "traces_host_info"
+	hostIdentifierAttr = "grafana.host.id"
+)
+
+var tracer = otel.Tracer("modules/generator/processor/hostinfo")
+
+type Processor struct {
+	Cfg    Config
+	logger log.Logger
+
+	gauge      registry.Gauge
+	registry   registry.Registry
+	metricName string
+}
+
+func (p *Processor) Name() string {
+	return Name
+}
+
+func (p *Processor) PushSpans(ctx context.Context, req *tempopb.PushSpansRequest) {
+	_, span := tracer.Start(ctx, "hostinfo.PushSpans")
+	defer span.End()
+
+	for i := 0; i < len(req.Batches); i++ {
+		resourceSpan := req.Batches[i]
+		attrs := resourceSpan.GetResource().GetAttributes()
+
+	idAttrLoop:
+		for _, idAttr := range p.Cfg.HostIdentifiers {
+			for _, attr := range attrs {
+				if attr.GetKey() == idAttr {
+					labelValues := p.registry.NewLabelValueCombo(
+						[]string{hostIdentifierAttr},
+						[]string{attr.GetValue().GetStringValue()},
+					)
+					p.gauge.Set(labelValues, 1)
```

Comment:
`.SetForTargetInfo` better matches the functionality you are trying to achieve here. Perhaps we should rename that method and use it here?

Reply:
`SetForTargetInfo` will pass `updateIfAlreadyExist` as false. This isn't the behavior we want as we intend to reset the `lastUpdated` value stored on the gauge every time a host identifier is matched.

I could add `SetForHostInfo` but it would essentially be a clone of `Set`.

Reply:
good catch. now i'm wondering why SetForTargetInfo exists? I feel like it might be a bug that it's not setting the lastUpdated field.

no need to make a `SetForHostInfo`. `Set` is fine.